source/Halibut.Tests/BaseTest.cs

```diff
@@ -31,11 +32,8 @@ public void SetUp()
         public void TearDown()
         {
             Logger.Information("Tearing down");
-            if (cancellationTokenSource != null)
-            {
-                cancellationTokenSource.Dispose();
-                cancellationTokenSource = null;
-            }
+            cancellationTokenRegistration?.Dispose();
```

Comment:
This was a leak resulting in 10's thousands of retained objects.

Reply:
Nice one